src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTableUI.java

```diff
@@ -592,19 +600,20 @@ private void paintCells(SynthContext context, Graphics g, int rMin,
                 }
             }
         } else {
-            for(int row = rMin; row <= rMax; row++) {
+            for (int row = rMin; row <= rMax; row++) {
                 cellRect = table.getCellRect(row, cMin, false);
-                aColumn = cm.getColumn(cMin);
+                cellRect.x = getXPosition(cMax);
+                aColumn = cm.getColumn(cMax);
                 if (aColumn != draggedColumn) {
                     columnWidth = aColumn.getWidth();
                     cellRect.width = columnWidth - columnMargin;
-                    paintCell(context, g, cellRect, row, cMin);
+                    paintCell(context, g, cellRect, row, cMax);
                 }
```

Comment:
I guess you removed this in BasicTableUI...Any particular reason for keeping this for Synth..